ModulBot/Controllers/MessageController.cs

```diff
@@ -0,0 +1,132 @@
+                    var indexOfSpace = str.IndexOf(" ", StringComparison.Ordinal);
+                    var command = indexOfSpace != -1 ? str.Substring(0, indexOfSpace).ToLower()
+                        : str.ToLower();
+                    if (command[0] != '/')
+                    {
+                        command = "/" + command;
+                    }
+                    if (command.Contains($"@{me.Username}"))
+                    {
+                        int startIndex = command.IndexOf('@');
+                        command = command.Substring(0, startIndex);
+                    }
+                    var help = new Help();
+
+                    var tool = Bot.Tools.FirstOrDefault(x => x.CommandsName.Any(y => y.Equals(command)));
+                    if (tool != null)
+                    {
+                        message.Text = indexOfSpace >= 0 ?
+                            message.Text.Substring(indexOfSpace, str.Length - indexOfSpace) : String.Empty;
+
+                        if (user == null || (!user.IsAdmin && (tool).IsAdmin))
+                        {
+                            help.Run(message);
+                            return;
+                        }
+                        tool.Run(message);
+                    }
+                    else
+                    {
+                        help.Run(message);
+                        return;
+                    }
+                }
+
+            }
+            catch
+            {
+                await Bot.BotClient.SendTextMessageAsync(update.CallbackQuery.Id,
+                    "Упс, что-то пошло не так. Бот не может обработать команду/нажатие" +
+                    " на кнопку. Но он продолжит работу");
+            }
+        }
+
+        public static async Task ButtonAction(string callbackInfo, long chatId, int messageId)
+        {
+            if (callbackInfo.Split(' ')[2].ToLower().Equals("false"))
+                for (int i = 0; i < Bot.NextButtons.Count; i++)
+                {
+                    if (Bot.NextButtons[i][0].CallbackData.Equals(callbackInfo))
+                    {
+                        await ActionButton.DoAction(chatId, messageId, callbackInfo.Split(' '));
+                        return;
+                    }
+                }
+            else
+                for (int i = 0; i < Bot.StatButtons.Count; i++)
+                {
+                    if (Bot.StatButtons[i][0].CallbackData.Equals(callbackInfo))
+                    {
+                        await ActionButton.ShowButtons(chatId, messageId, callbackInfo.Split(' '));
+                        return;
+                    }
+                }
+            try
+            {
+                await Bot.BotClient.SendTextMessageAsync(chatId,
+                    "Попробуйте заново ввести команду /tunestat(/statonoff - для админа)");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS2228)